apps/api/src/Command/GreencheckDataDumpCommand.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace App\Command;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Predis\Client;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use TGWF\Greencheck\Sitecheck;
+
+/**
+ * GreencheckDataDumpCommand.
+ *
+ * Dump all redis result keys for domains
+ */
+class GreencheckDataDumpCommand extends Command
+{
+    protected $checker = null;
+    protected $output = null;
+
+    /**
+     * @var ParameterBagInterface
+     */
+    private $params;
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    public function __construct(
+        $name = null,
+        ParameterBagInterface $params,
+        Client $client
+    ) {
+        parent::__construct($name);
+        $this->params = $params;
+        $this->client = $client;
+    }
+
+    /**
+     * Configure the command.
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('tgwf:greencheck:datadump')
+            ->setDescription('Make a csv database dump from redis')
+            ;
+    }
+
+    /**
+     * Execute the command.
+     *
+     * @param InputInterface  $input  Input
+     * @param OutputInterface $output Output
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $fp = fopen('tgwf-dump.csv', 'w');
+
+        $first = true;
+        $allDomainKeys = $this->client->keys('domains:*');
+
+        // creates a new progress bar (50 units)
+        $progressBar = new ProgressBar($output, count($allDomainKeys));
+
+        // starts and displays the progress bar
+        $progressBar->start();
+
+        foreach($allDomainKeys as $key) {
```

Comment:
Hey @arendjantetteroo  i don't know enough about PHP to answer this myself, but if I'm looking through a big collection in python or javascript, I'll rely on a generator, so I don't have to keep the whole list of objects in memory, and load them in bunches of a few hundred or thousand to avoid exhausting the memory.

If we have a few million results here, would we be loading a few million results into memory in this case before looping in them?

Or is there some clever loading magic going on under the hood in PHP that takes care of this for us?

Reply:
php has generators as well, i'm not sure if the redis method uses them for the $allDomainKeys. 
Best would be to just try it out. As we run it on cli and have loads of ram in the server box, i think we will be fine.

Reply:
heh, works for me :)